pkg/nsx/services/subnet/builder.go

```diff
@@ -91,7 +91,14 @@ func (service *SubnetService) buildSubnet(obj client.Object, tags []model.Tag, i
 		if dhcpMode == "" {
 			dhcpMode = v1alpha1.DHCPConfigModeDeactivated
 		}
-		nsxSubnet.SubnetDhcpConfig = service.buildSubnetDHCPConfig(dhcpMode, nil)
+		var dhcpServerAdditionalConfig *model.DhcpServerAdditionalConfig
+		if o.Spec.SubnetDHCPConfig.DHCPServerAdditionalConfig.ReservedIPRanges != nil && len(o.Spec.SubnetDHCPConfig.DHCPServerAdditionalConfig.ReservedIPRanges) > 0 {
```

Comment:
```suggestion
		if len(o.Spec.SubnetDHCPConfig.DHCPServerAdditionalConfig.ReservedIPRanges) > 0 {
```
as len() supports nil array?

Reply:
make sense. Changed.